src/backend/sycl_backend.cc

```diff
@@ -9,6 +9,14 @@
 
 #include <fmt/ranges.h>
 
+namespace {
+void wait_synchronously_for(const celerity::detail::async_event& ev) {
```

Comment:
I don't believe you have to wait for any of the events below. They are executed in-order with other submissions to the worker thread anyway.